src/components/dev-hub/star-rating.js

```diff
@@ -0,0 +1,180 @@
+import React, { useCallback, useReducer } from 'react';
+import PropTypes from 'prop-types';
+import StarIcon from '~components/dev-hub/icons/star-icon';
+import styled from '@emotion/styled';
+import { size, fontSize, screenSize } from './theme';
+
+const StyledContainer = styled.div`
+    display: flex;
+    flex-direction: column;
+
+    @media ${screenSize.mediumAndUp} {
+        flex-direction: row;
+    }
+`;
+
+const StyledStar = styled(StarIcon)`
+    width: ${size.mediumLarge};
+    height: ${size.mediumLarge};
+    margin-right: ${size.mediumLarge};
+    box-sizing: content-box;
+
+    @media ${screenSize.mediumAndUp} {
+        padding: 4px;
+        width: auto;
+        height: auto;
+        margin-right: ${({ isActive }) => (isActive ? '0' : size.xsmall)};
+
+        &:hover {
+            cursor: ${({ isClicked }) => (isClicked ? 'default' : 'pointer')};
+        }
+    }
+`;
+
+const StyledStarsContainer = styled.div`
+    display: flex;
+    align-items: center;
+
+    @media ${screenSize.mediumAndUp} {
+        margin-left: ${size.large};
+    }
+`;
+
+const StyledTitle = styled.span`
+    font-family: 'Fira Mono', monospace;
+    font-weight: bold;
+    font-size: ${fontSize.small};
+`;
+
+const initialState = {
+    first: false,
+    second: false,
+    third: false,
+    fourth: false,
+    fifth: false,
+    isClicked: false,
+};
+
+const reducer = (state, action) => {
+    let nextState = { ...initialState };
+    switch (action) {
+        case 'fifth':
+            nextState = { ...nextState, fifth: true };
+        case 'fourth':
+            nextState = { ...nextState, fourth: true };
+        case 'third':
+            nextState = { ...nextState, third: true };
+        case 'second':
+            nextState = { ...nextState, second: true };
+        case 'first':
+            nextState = { ...nextState, first: true };
```

Comment:
I like this approach to adding in first, second, third, etc.! Nice work

A small performance optimization might be to only spread `nextState` at the end instead of (potentially) 5 times (one for each switch case). We could start a new object which we add to for making `fifth` `true`, and `fourth` `true`, etc.

Then at the end we would return a new object which spreads `nextState` and this object we incrementally build. Thoughts?

Reply:
We could also use an array as well to reduce the number of keys in our object and be able to scale to different numbers of stars.